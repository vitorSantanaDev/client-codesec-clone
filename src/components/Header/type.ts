export type IViewProps = {
  logo: string
  logoAltText: string
  buttontext: string
}
