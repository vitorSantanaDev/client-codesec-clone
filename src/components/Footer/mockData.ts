export const mockDataFooterProps = {
  logo: {
    altText: 'logo codesec',
    urlImage: '/images/logo-2-codesec.svg'
  },
  footerContent: 'Todos od direitos reservados',
  links: [
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/instagram.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/facebook.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/youtube.svg'
      }
    }
  ]
}
