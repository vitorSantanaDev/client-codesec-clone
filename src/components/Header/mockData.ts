import { IViewProps } from './type'

export const mockHeaderProps: IViewProps = {
  logo: {
    urlImage: '/images/codesec-logo.svg',
    altText: 'Logo codesec'
  },
  button: {
    url: '/',
    buttonText: 'Inscreva-se'
  }
}
