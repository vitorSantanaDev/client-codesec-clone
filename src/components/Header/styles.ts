import styled, { css } from 'styled-components'
import { Logo as LogoElement } from 'components/Logo/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    width: 100%;
    padding: calc(${theme.spacings.small} - 0.6rem);

    z-index: 200;
    position: fixed;

    display: flex;
    align-items: center;
    justify-content: space-between;

    border: 1px solid ${theme.colors.blackLight};

    @supports (backdrop-filter: blur(16px)) {
      background: rgba(10, 9, 9, 0.3);
      backdrop-filter: blur(16px);
    }

    ${LogoElement} {
      width: 220px;
    }

    .logo {
      @media ${theme.media.lteMedium} {
        width: 145px;
      }
    }
  `}
`
