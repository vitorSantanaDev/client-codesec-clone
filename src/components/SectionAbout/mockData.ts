export const mockSectionAboutProps = {
  subtitle: 'infos',
  title: 'About web series',
  description:
    'Aprenda PHP do Jeito Certo Utilizando 100% de Tecnologia Moderna e os Melhores Padr√µes de Projeto Nessa Webserie 100% Online e GRATUITA!',
  topics: [
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    }
  ],
  illustration: {
    urlImage: '/images/about-ilustration.svg',
    altText: 'Illustration about'
  }
}
