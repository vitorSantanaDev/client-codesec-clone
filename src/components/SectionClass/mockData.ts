export const mockDataSectionClassProps = {
  title: 'O que você vai aprender',
  subtitle: 'Aulas',
  classs: [
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    }
  ]
}
