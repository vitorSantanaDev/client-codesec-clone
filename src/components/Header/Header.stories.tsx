import { ComponentStory, ComponentMeta } from '@storybook/react'
import Header from '.'
import { mockHeaderProps } from './mockData'

export default {
  title: 'Header',
  component: Header,
  args: {
    ...mockHeaderProps
  }
} as ComponentMeta<typeof Header>

export const Default: ComponentStory<typeof Header> = (args) => (
  <Header {...args} />
)
