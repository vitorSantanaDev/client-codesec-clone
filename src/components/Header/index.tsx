import Button from 'components/Button'
import Logo from 'components/Logo'

import { IViewProps } from './type'
import * as S from './styles'

const Header = ({ logo, button }: IViewProps) => (
  <S.Wrapper>
    <Logo
      className="logo"
      url="/"
      srcImage={logo.urlImage}
      altText={logo.altText}
    />
    <Button size="small">{button.buttonText}</Button>
  </S.Wrapper>
)

export default Header
