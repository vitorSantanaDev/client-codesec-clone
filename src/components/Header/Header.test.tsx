import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'

import Header from '.'
import { mockHeaderProps } from './mockData'

describe('<Header />', () => {
  it('should render Header with correct elements', () => {
    const { container } = renderTheme(<Header {...mockHeaderProps} />)
    const button = screen.getByRole('button', { name: /inscreva-se/i })
    const logo = screen.getByRole('img', { name: /logo/i })

    expect(logo).toBeInTheDocument()
    expect(button).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render Header with correct styles', () => {
    const { container } = renderTheme(<Header {...mockHeaderProps} />)
    expect(container.firstChild).toHaveStyle({
      'z-index': '200',
      position: 'fixed'
    })
    expect(container.firstChild).toMatchSnapshot()
  })
})
