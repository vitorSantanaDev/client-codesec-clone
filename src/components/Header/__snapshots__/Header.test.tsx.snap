// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header with correct elements 1`] = `
.c3 {
  width: 100%;
  max-width: 190px;
  padding: calc(1.6rem - 0.4rem);
  font-weight: 600;
  text-transform: unset;
  font-size: 1.6rem;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  color: #FFFFFF;
  background: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  -webkit-transition: all ease-in-out 300ms;
  transition: all ease-in-out 300ms;
}

.c3:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  box-shadow: -10px 15px 40px -15px #FA5770;
}

.c2 {
  cursor: pointer;
}

.c0 {
  width: 100%;
  padding: calc(1.6rem - 0.6rem);
  z-index: 200;
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border: 1px solid #1C1B1B;
}

.c0 .c1 {
  width: 220px;
}

@supports (-webkit-backdrop-filter:blur(16px)) or (backdrop-filter:blur(16px)) {
  .c0 {
    background: rgba(10,9,9,0.3);
    -webkit-backdrop-filter: blur(16px);
    backdrop-filter: blur(16px);
  }
}

@media (max-width:768px) {
  .c0 .logo {
    width: 145px;
  }
}

<header
  class="c0"
>
  <div
    class=""
  >
    <img
      alt="Logo codesec"
      class="c1 c2 logo"
      src="/images/codesec-logo.svg"
    />
  </div>
  <button
    class="c3"
  >
    Inscreva-se
  </button>
</header>
`

exports[`<Header /> should render Header with correct styles 1`] = `
.c3 {
  width: 100%;
  max-width: 190px;
  padding: calc(1.6rem - 0.4rem);
  font-weight: 600;
  text-transform: unset;
  font-size: 1.6rem;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  color: #FFFFFF;
  background: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  -webkit-transition: all ease-in-out 300ms;
  transition: all ease-in-out 300ms;
}

.c3:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  box-shadow: -10px 15px 40px -15px #FA5770;
}

.c2 {
  cursor: pointer;
}

.c0 {
  width: 100%;
  padding: calc(1.6rem - 0.6rem);
  z-index: 200;
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border: 1px solid #1C1B1B;
}

.c0 .c1 {
  width: 220px;
}

@supports (-webkit-backdrop-filter:blur(16px)) or (backdrop-filter:blur(16px)) {
  .c0 {
    background: rgba(10,9,9,0.3);
    -webkit-backdrop-filter: blur(16px);
    backdrop-filter: blur(16px);
  }
}

@media (max-width:768px) {
  .c0 .logo {
    width: 145px;
  }
}

<header
  class="c0"
>
  <div
    class=""
  >
    <img
      alt="Logo codesec"
      class="c1 c2 logo"
      src="/images/codesec-logo.svg"
    />
  </div>
  <button
    class="c3"
  >
    Inscreva-se
  </button>
</header>
`
